lib/Analysis/Allocation.cpp

```diff
@@ -135,10 +135,19 @@ getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
   // TODO(jlebar): This is suboptimal if repShape[in/outOrd[0]] is small.  We
   // might be able to merge the few most-minor dimensions and get a larger
   // vector.
-  // For conversions to MmaV1 (Nvidia V100), this inVec is hardcoded in the
-  // codegen.
   inVec = std::min(srcContigPerThread[inOrd[0]], dstContigPerThread[inOrd[0]]);
   outVec = dstContigPerThread[outOrd[0]];
+
+  if (srcLayout.isa<BlockedEncodingAttr>() &&
+      dstLayout.isa<BlockedEncodingAttr>() && outOrd[0] == 0 && inOrd[0] != 0) {
+    // don't vectorize for transpose as only the read or the write can be
+    // vectorized and this cause extra bank conflicts on the non-vectorized
+    // access.
```

Comment:
Hm, I don't understand what the problem is here.

For example, suppose we're doing a 32x32 shmem transpose.  Suppose the reads can be vectorized but the writes can't be.  How does vectorizing the reads cause bank conflicts, exactly?  All of the writes complete before any of the reads finish, so how do they conflict?

Reply:
so if we have a 32x32 transpose, if want to avoid bank conflicts with scalar load and stores we would want to pad by 1 element. If we vectorize the load to a load4 we can only pad with 0 or 4 elements. 4 would mean no conflicts on read but extra conflicts on write, 0 would mean conflicts on both.

Reply:
Ahhhh, I see.  Could you update the comment to say something like this?

Reply:
sure, let me try

Reply:
added this example in the comment